bindings/haskell/haskell-src/OpenDAL.hs

```diff
@@ -63,6 +67,28 @@ data ErrorCode
 data OpenDALError = OpenDALError {errorCode :: ErrorCode, message :: String}
   deriving (Eq, Show)
 
+data EntryMode = File | Dir | Unknown deriving (Eq, Show)
+
+data BytesContentRange = BytesContentRange
+  { bcrStart :: Integer
+  , bcrEnd :: Integer
+  , bcrTotal :: Integer
+  }
+  deriving (Eq, Show)
+
+data Metadata = Metadata
+  { mMode :: EntryMode
+  , mCacheControl :: Maybe String
+  , mContentDisposition :: Maybe String
+  , mContentLength :: Integer
```

Comment:
Does `Integer` here contains u64?

Reply:
Yes. There are only `Int` and `Integer` in haskell. `Int` stands `i32`, and `Interger` stands an infinite number.

Reply:
Cool, lesson learnt.